static/docs/commands-reference/add.md

```diff
@@ -2,82 +2,193 @@
+detailed description of the DVC _metafile_ format.
+
+By default DVC tries a range of link types (`reflink`, `hardlink`, `symlink`,
+or `copy`) to try to avoid copying any file contents and to optimize DVC file 
+operations even for large files. The `reflink` is the best link type available, 
+but even though it is frequently supported by modern filesystems, many others 
+still don't support it.  DVC has the other link types for use on filesystems 
+without `reflink` support. See `dvc config` for more information.
+
+A `dvc add` target can be an individual file or a directory.  There are two ways
+to work with directory hierarchies with `dvc add`.
+
+1. With `dvc add --recursive`, the hierarchy is traversed and every file is 
+  added individually as described above.  This means every file has its own 
+  `.dvc` file, and a corresponding DVC cache entry is made.  If the 
+  `--no-commit` flag is added the DVC cache entry is not made.
+2. When not using `--recursive` a DVC stage file is created for the top of 
+  the directory (`dirname.dvc`), and every file in the hierarchy is added to the 
+  DVC cache, but these files do not have individual DVC files.  Instead the DVC 
+  file for the directory has a corresponding file in the DVC cache containing 
+  references to the files in the directory hierarchy.
+
+The `dvc add` command is useful for manually maintaining updates to files.  For
+files that are the outputs of running a command, it is better to use the 
+`dvc run` command to create a DVC stage file listing dependencies and outputs.
+
+## Options
+
+* `-R`, `--recursive`  Recursively add each file under the named directory.  For 
+  each file a new DVC file is created using the process described earlier.
+
+* `--no-commit` Do not put files/directories into cache. A stage file is created,
+  and an entry is added to `.dvc/state`, while nothing is added to the
+  cache (`.dvc/cache`).  The `dvc status` command will note that the file
+  is `not in cache`.  The `dvc commit` command will add the file to 
+  the DVC cache.  This is analogous to the `git add` and `git commit` commands.
+
+* `-h`, `--help` - prints the usage/help message, and exit.
+
+* `-q`, `--quiet` - do not write anything to standard output.  Exit with 0 if
+  all stages are up to date or if all stages are successfully rerun, otherwise 
+  exit with 1.
+
+* `-v`, `--verbose` - displays detailed tracing information from executing the
+  `dvc add` command.
 
-For directories, the command does the same steps for each file recursively.
-To retain information about the directory structure, a corresponding cache
-file will be created in `.dvc/cache`.
 
 ## Examples
 
 Take files under DVC control:
 
 ```dvc
+
     $ ls raw
 
-    Badges.xml          PostLinks.xml           Votes.xml
+    dog.111.jpg	dog.121.jpg	dog.131.jpg	dog.141.jpg
+
+    $ dvc add raw/dog.111.jpg raw/dog.121.jpg raw/dog.131.jpg raw/dog.141.jpg
```

Comment:
it's very uncommon to put 4 images separately. Either let's do a directory, or a single data file. Overall, let's simplify again. A lot of these long outputs below could me omitted.